flang/lib/Parser/openmp-parsers.cpp

```diff
@@ -804,6 +823,10 @@ TYPE_PARSER(sourced(construct<OmpSimpleStandaloneDirective>(first(
 TYPE_PARSER(sourced(construct<OpenMPSimpleStandaloneConstruct>(
     Parser<OmpSimpleStandaloneDirective>{}, Parser<OmpClauseList>{})))
 
+// 14.1 Interop construct
+TYPE_PARSER(sourced(construct<OpenMPInteropConstruct>(
```

Comment:
Okay, I see you have added `sourced(` here, do the same for the leaf nodes as well.